utils/buffer/src/buffer_pool/mod.rs

```diff
@@ -110,25 +178,54 @@ impl PoolFront {
     }
 }
 
+// Current mode of operation for the `BufferPool`.
+//
+// The pool operates in three modes based on memory availability: it first allocates from the back,
+// then from the front if the back is full, and finally from system memory (with reduced
+// performance) if both sections are exhausted.
```

Comment:
Since this `pub enum PoolMode` is a field of the public struct `BufferPool`, shouldn't it have public doc comments? 

I see it when I render docs locally:
<img width="1218" alt="Screenshot 2024-12-10 at 09 26 07" src="https://github.com/user-attachments/assets/017b1eb7-a689-45d5-842d-7460daa5d294">